lib/matplotlib/backend_tools.py

```diff
@@ -36,6 +37,35 @@ class Cursors(enum.IntEnum):  # Must subclass int for the macOS backend.
     RESIZE_VERTICAL = enum.auto()
 cursors = Cursors  # Backcompat.
 
+
+# _tool_registry, _register_tool_class, and _find_tool_class implement a
+# mechanism through which ToolManager.add_tool can determine whether a subclass
+# of the requested tool class has been registered (either for the current
+# canvas class or for a parent class), in which case that tool subclass will be
+# instantiated instead.  This is the mechanism used e.g. to allow different
+# GUI backends to implement different specializations for ConfigureSubplots.
+
+
+_tool_registry = set()
+
+
+def _register_tool_class(canvas_cls, tool_cls=None):
```

Comment:
I don't see how to do it easily, but I still need to ask, is it possible to add test to these two functions?

Reply:
They are covered already by existing tests (https://codecov.io/gh/matplotlib/matplotlib/compare/971d1106464560038151bf79d032515cd843b149...49d855f552bb44282b40ff689c5de5fe60ffbc15/diff).